### ISSUE TITLE
add middleware/startup and middleware/shutdown

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#600]
See
https://caddyserver.com/docs/shutdown

and

https://caddyserver.com/docs/shutdown

**Comment by miekg** [comment]
This is a one-liner:

~~~
diff --git a/middleware.cfg b/middleware.cfg
index 75985cf..4a30a87 100644
--- a/middleware.cfg
+++ b/middleware.cfg
@@ -42,3 +42,4 @@
200:proxy:proxy
210:whoami:whoami
220:erratic:erratic
+500:startup:github.com/mholt/caddy/startupshutdown
~~~
/ht @mholt for making me try it

**Comment by miekg** [comment]
We already depend on caddy, so this is basically a no-brainer :) I will submit a PR shortly

**Comment by miekg** [comment]
closed with  830fdfd
